src/pydvl/utils/types.py

```diff
@@ -92,3 +95,26 @@ def __call__(cls, *args, **kwargs):
 
     def create(cls, *args: Any, **kwargs: Any):
         return super().__call__(*args, **kwargs)
+
+
+Seed = Union[int, Generator]
+
+
+def ensure_seed_seq(
+    seed: Optional[int, Generator, SeedSequence] = None
+) -> SeedSequence:
+    """
+    If the passed seed is a SeedSequence object then it is returned as is. If it is
+    a Generator the internal protected seed sequence from the generator gets extracted.
+    Otherwise, a new SeedSequence object is created from the passed (optional) seed.
+
+    :param seed: Either an int, a Generator object a SeedSequence object or None.
+
+    :returns: A SeedSequence object.
+    """
+    if isinstance(seed, SeedSequence):
+        return seed
+    elif isinstance(seed, Generator):
+        return seed._bit_generator._seed_seq  # noqa
```

Comment:
One fewer private member access:

```suggestion
        return seed.bit_generator._seed_seq  # noqa
```

https://numpy.org/doc/1.20/reference/random/generated/numpy.random.Generator.bit_generator.html#numpy.random.Generator.bit_generator